src/interpreter/crypto.rs

```diff
@@ -59,12 +61,15 @@ impl<S> Interpreter<S> {
     }
 
     pub(crate) fn sha256(&mut self, a: Word, b: Word, c: Word) -> Result<(), RuntimeError> {
-        if a > VM_MAX_RAM - Bytes32::LEN as Word || c > MEM_MAX_ACCESS_SIZE || b > VM_MAX_RAM - c {
+        if a > VM_MAX_RAM.saturating_sub(Bytes32::LEN as Word)
+            || c > MEM_MAX_ACCESS_SIZE
+            || b > VM_MAX_RAM.saturating_sub(c)
+        {
             return Err(PanicReason::MemoryOverflow.into());
         }
 
         let (a, b, c) = (a as usize, b as usize, c as usize);
-        let bc = b + c;
+        let bc = b.saturating_add(c);
 
         self.try_mem_write(a, Hasher::hash(&self.memory[b..bc]).as_ref())?;
```

Comment:
Consider additionally using `.get()` instead of directly slicing to avoid potential rust level panics.

```suggestion
        self.try_mem_write(a, Hasher::hash(&self.memory.get(b..bc).ok_or(.. some bug error type ..)).as_ref())?;
```

Reply:
We need to keep in mind that 95% of these cases are not runtime, such as https://github.com/FuelLabs/fuel-vm/pull/203/files#diff-df812721f376b9b2f9599469028b29653b7341d65fa823c5b4d8bfd2127d5032R11

In fact, using checked arith there is maybe even overkill. Its just applied because clippy isn't smart enough to differentiate between what is fixed by the compiler and what is resolved at runtime. I'll recheck the ones that are runtime and might overflow